### PR TITLE
Feature/optimize for polling and attachment handler

### PR DESCRIPTION
**Pull request by lucidNTR**

### CODE REVIEW

**Comment by lucidNTR** [comment]
@gr2m

File: package.json [hunk 1]
```diff
@@ -29,19 +31,30 @@
...
+  },
+  "maintainers": [
+    {
+      "name": "gr2m",
+      "email": "gregor@martynus.net"
+    }
+  ],
+  "publishConfig": {
+    "registry": "https://npme.eocng.org/",
+    "tag": "next"
```
Comment (gr2m) [inline package.json]: why do you want to release it on our on-premise npm registry? I wouldn't mind moving the repo from gr2m to eHealtAfrica, and keep the repo open source? Or is there anything eHealth specific in it?

File: .travis.yml [hunk 2]
```diff
@@ -1,3 +1,22 @@
...
+cache:
+  directories:
+    - node_modules
+notifications:
+  email: false
 node_js:
-  - "0.10"
+  - 4
+before_install:
+  - echo "//npme.eocng.org/:_authToken=$NPM_TOKEN" >> .npmrc
```
Comment (gr2m) [inline .travis.yml]: you can leave out this line, semantic-release does it automatically by now

File: .travis.yml [hunk 3]
```diff
@@ -1,3 +1,22 @@
...
+  - npm install -g npm@^3.0.0
+install:
+  - npm install
+before_script:
+  - npm prune
+after_success:
+  - npm run semantic-release && scripts/docker-build.sh
```
Comment (gr2m) [inline .travis.yml]: what is `scripts/docker-build.sh`? It's not in the code?

File: package.json [hunk 4]
```diff
@@ -29,19 +31,30 @@
...
+    "sinon": "^1.12.2",
+    "semantic-release": "^4.3.5",
+    "standard": "^5.3.1"
   },
   "dependencies": {
     "async": "^0.9.0",
-    "debounce": "^1.0.0",
-    "mail-listener2": "git://github.com/gr2m/mail-listener2.git",
+    "imap": "^0.8.15",
+    "underscore": "^1.8.3",
```
Comment (gr2m) [inline package.json]: I'd avoid underscore if possible, due to its breaking releases in the past

**Comment by gr2m** [comment]
once we merged this, let's move the main repo over to https://github.com/eHealthAfrica/download-email-attachments

File: package.json (hunk 1 above)
Comment (lucidNTR) [inline package.json]: that was the plan, i just did not know if we publish always to on-premise and mirror to public npm

File: package.json (hunk 4 above)
Comment (lucidNTR) [inline package.json]: lodash is ok?

File: package.json (hunk 1 above)
Comment (gr2m) [inline package.json]: Ah okay, that is a very good point, we should document that. We don't do any mirroring, we only publish our private modules to our on-premise npm registry

File: package.json (hunk 4 above)
Comment (gr2m) [inline package.json]: yes lodash is perfect. With lodash you can also only require what you need: https://www.npmjs.com/package/lodash.after

File: package.json (hunk 1 above)
Comment (lucidNTR) [inline package.json]: removed private publishing

File: .travis.yml (hunk 2 above)
Comment (lucidNTR) [inline .travis.yml]: removed

File: .travis.yml (hunk 3 above)
Comment (lucidNTR) [inline .travis.yml]: removed. no idea, accidentally copied over from other repo...

**Pull request merged**